### ISSUE TITLE
handleButtonAction() and checkDisabled(item) do not work for actionButton

### ISSUE THREAD

**Issue by AparnaKarve** [patternfly/angular-patternfly#326]
I have a use case where I need to -
(a) Disable the actionButton the moment it's clicked
(b) Change the actionButton text when it's clicked

For e.g. -- In the screenshot below, there is a `Start` actionButton for each Service.
When I click on Start, I need the button to be disabled right away, and I also need to update the text from `Start` to `Starting`

<img width="1387" alt="screen shot 2016-10-12 at 10 52 04 am" src="https://cloud.githubusercontent.com/assets/1538216/19321645/b94d65ce-906a-11e6-9a44-d4b83019ed4e.png">
## Issues  with actionButton

The html generated for the actionButton is as shown below.
- `handleButtonAction` and `checkDisabled(item)` functions do not seem to work
- Tooltip on the button shows `"actionButton.title"`

``` html
<button class="btn btn-default ng-binding ng-scope btn-primary" ng-repeat="actionButton in actionButtons" title="actionButton.title" ng-class="{'disabled' : checkDisabled(item) || !enableButtonForItem(actionButton, item)}" ng-click="handleButtonAction(actionButton, item)">Start</button>
```

**Comment by AparnaKarve** [comment]
/cc @jeff-phillips-18 @dtaylor113 

It would be great if we could fix this as well for the next angular-patternfly release.

**Comment by jeff-phillips-18** [comment]
the handleButtonAction and and checkDisabled are both working on the ngDocs page and in the unit tests. Do you have some example code that shows this not working?

Agreed, the tooltip is incorrect.

**Comment by jeff-phillips-18** [comment]
For your use case I was able to accomplish this with:

        var startButtonInclude = '<span>{{item.started ? "Starting" : "Start"}}</span>';
        $templateCache.put('start-button-template', startButtonInclude);

        var startServer = function (action, item) {
          $scope.eventText = item.name + " : " + action.name + "\r\n" + $scope.eventText;
          item.started = true;
        };

        $scope.enableButtonForItemFn = function(action, item) {
          return !(action.name === 'Start' && item.started);
        };


 Start Action Button Definition:
          {
            name: 'Start',
            class: 'btn-primary',
            include: 'start-button-template',
            title: 'Start the server",
            actionFn: startServer
          },

**Comment by AparnaKarve** [comment]
@jeff-phillips-18 I have uploaded a remote branch on the `manageiq-ui-self_service` repo here that's pretty much my example code:
https://github.com/ManageIQ/manageiq-ui-self_service/compare/master...AparnaKarve:power_ops?expand=1

With your above suggested changes, all Services display `Start` and not `Starting` (see screenshot)

<img width="1394" alt="screen shot 2016-10-12 at 1 22 33 pm" src="https://cloud.githubusercontent.com/assets/1538216/19326370/1a7835f4-907f-11e6-90dd-bd5f22c5b923.png">

I have currently mocked the data for `item.startService`, so the Start button for each Service is either enabled or disabled. I want the enabled ones to say `Start` and the disabled ones to say `Starting`

**Comment by jeff-phillips-18** [comment]
You will need angular patternfly 3.12

For your start button definition you will need to specify:
  include: 'start-button-template',

And in the startService method:
  item.startService = false;

**Comment by AparnaKarve** [comment]
Ok, cool. In that case, I'm very much looking forward to 3.12 😄

**Comment by dtaylor113** [comment]
This appears this was fixed in angular-patternfly v3.12.
